### PR TITLE
Add a date_option (format string) parameter to the choropleth timeline slider

### LINKED ISSUE #1997: Format time to show year only in choropleth time slider
There may already be a way to do this, but I was not able to stumble on it...

I would like to be able to control the data format in the slider bar in a time slider choropleth.  In my case (and I believe many others), I am describing annual trends and all that is required in the slider is the year.

I am requesting:
1) Ability to set the time format displayed in the choropleth time slider itself by passing a date format. or setting a "show year only" variable, or if this is already present,
2) Enhanced documentation describing how this can be modified.

### PR DESCRIPTION
**Pull request by hansthen**
Closes #1997

### CODE REVIEW

File: folium/plugins/time_slider_choropleth.py [hunk 1]
```diff
@@ -19,6 +19,8 @@ class TimeSliderChoropleth(JSCSSMixin, Layer):
     styledict: dict
         A dictionary where the keys are the geojson feature ids and the values are
         dicts of `{time: style_options_dict}`
+    date_options: str, default "ddd MMM DD YYYY"
```
Comment (Conengmo) [inline folium/plugins/time_slider_choropleth.py:22]: later on it's called `date_format`, why not call it that here as well?

File: folium/plugins/time_slider_choropleth.py [hunk 2]
```diff
@@ -19,6 +19,8 @@ class TimeSliderChoropleth(JSCSSMixin, Layer):
...
+        A format string to render the currently active time in the control.
```
Comment (Conengmo) [inline folium/plugins/time_slider_choropleth.py:23]: should we specify what formatting language is being used here? Like, what fields like MMM DD are available?

**Review by Conengmo — COMMENTED** [review]
Looks good! Two comments that are more questions/suggestions on how to improve, but not critical.

File: folium/plugins/time_slider_choropleth.py (hunk 1 above)
Comment (hansthen) [reply folium/plugins/time_slider_choropleth.py:22]: To be consistent with the other plugins. `timestamped_geojson` and `timeline` also use `date_options`. I'm not so happy about the name

File: folium/plugins/time_slider_choropleth.py (hunk 2 above)
Comment (hansthen) [reply folium/plugins/time_slider_choropleth.py:23]: If you feel strongly about it, I can add a link to the formatting library used. However, I don't think it is necessary. The time string formatting language is fairly intuitive (and fairly universal). Also, I would like the documentation to be consistent between the different plugins.

File: folium/plugins/time_slider_choropleth.py (hunk 1 above)
Comment (Conengmo) [reply folium/plugins/time_slider_choropleth.py:22]: That makes sense 👍

**Review by Conengmo — APPROVED** [review]

**Pull request merged**